ports.go

```diff
@@ -26,140 +28,105 @@ type DevicePortServiceOp struct {
+	device, _, err := i.client.Devices.Get(deviceID)
 	if err != nil {
-		return nil, resp, false, err
+		return nil, err
+	}
+	for _, port := range device.NetworkPorts {
+		if port.Name == name {
+			return &port, nil
+		}
 	}
 
-	return unassignResponse, resp, true, err
+	return nil, fmt.Errorf("Port %s not found in device %s", name, deviceID)
 }
 
-type PortBondRequest struct {
-	PortID     string
-	BulkEnable bool
+func (i *DevicePortServiceOp) Assign(portID, vlanID string) (*Port, *Response, error) {
+	path := fmt.Sprintf("%s/%s/assign", portBasePath, portID)
+	return i.assignmentAction(portID, vlanID, path)
 }
 
-func (i *DevicePortServiceOp) Bond(portID string, bulkEnable bool) (*Port, *Response, error) {
-	path := fmt.Sprintf("%s/%s/bond", portBasePath, portID)
-	if bulkEnable {
-		path += "?bulk_enable=true"
-	}
-	output := new(Port)
-
-	resp, err := i.client.DoRequest("POST", path, nil, output)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return output, resp, nil
+func (i *DevicePortServiceOp) Unassign(portID, vlanID string) (*Port, *Response, error) {
+	path := fmt.Sprintf("%s/%s/unassign", portBasePath, portID)
+	return i.assignmentAction(portID, vlanID, path)
 }
 
-type PortDisbondRequest struct {
-	PortID      string
-	BulkDisable bool
+func (i *DevicePortServiceOp) assignmentAction(portID, vlanID, path string) (*Port, *Response, error) {
+	req := PortRequest{
+		PortID:           portID,
+		VirtualNetworkID: vlanID,
+	}
+	return i.portAction(path, &req)
 }
 
-func (i *DevicePortServiceOp) Disbond(portID string, bulkDisable bool) (*Port, *Response, error) {
+func (i *DevicePortServiceOp) Bond(portID string, bulkEnable bool) (*Port, *Response, error) {
 	path := fmt.Sprintf("%s/%s/bond", portBasePath, portID)
-	if bulkDisable {
-		path += "?bulk_disable=true"
-	}
-	output := new(Port)
-
-	resp, err := i.client.DoRequest("POST", path, nil, output)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return output, resp, nil
+	req := BondRequest{PortID: portID, BulkEnable: bulkEnable}
+	return i.portAction(path, &req)
 }
 
-func (i *DevicePortServiceOp) GetBondedPort(deviceID string) (*Port, bool, error) {
-	device, _, err := i.client.Devices.Get(deviceID)
-	for _, port := range device.NetworkPorts {
-		if port.Type == "NetworkBondPort" {
-			return &port, true, nil
-		}
-	}
-
-	return nil, false, err
+func (i *DevicePortServiceOp) Disbond(portID string, bulkDisable bool) (*Port, *Response, error) {
+	path := fmt.Sprintf("%s/%s/disbond", portBasePath, portID)
+	req := DisbondRequest{PortID: portID, BulkDisable: bulkDisable}
+	return i.portAction(path, &req)
 }
 
-// Private helper methods
-func (i *DevicePortServiceOp) assignVirtualNetwork(input *PortAssignRequest) (*PortAssignResponse, *Response, bool, error) {
-	path := fmt.Sprintf("%s/%s/assign", portBasePath, input.PortID)
-	assignResponse := new(PortAssignResponse)
+func (i *DevicePortServiceOp) portAction(path string, req interface{}) (*Port, *Response, error) {
```

Comment:
By moving the struct abstraction to the internals of the API call, you're removing the reason to use a struct in the first place: flexibility and ease of refactoring required arguments later. If you need to add an additional argument to any of these API calls, you will not only have to refactor all the code that calls these functions, but also have to refactor the struct to add an additional field, and unpack the arguments into the struct within the function.

Reply:
I put it back to structs in 3c2b87a9dfde24d7d7f1126ee55aaa5ae8fee390